sway-core/src/type_system/info.rs

```diff
@@ -870,13 +870,22 @@ impl GetCallPathWithEngines for TypeInfo {
             TypeInfo::RawUntypedPtr => None,
             TypeInfo::RawUntypedSlice => None,
             TypeInfo::Ptr(generic_argument) => generic_argument
-                .call_path_tree()
+                .as_type_argument()
```

Comment:
That repeating pattern mention before is again present here in many files many times, but slightly different with `map` this time and not `cloned`. The number of occurrences of the pattern just convince me more that we should abstract it in a method.